launch_ros/launch_ros/actions/explicit_ros_startup.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018-2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+from typing import Optional
+
+import launch.actions
+import launch_ros.ros_adapters
+
+import rclpy
+
+
+class ExplicitROSStartup(launch.actions.OpaqueFunction):
```

Comment:
I'm guessing that the purpose of this action is to allow users to initialize the ROS adapter. But, I don't think we should encourage users to use the launch ROS context (or node) directly. It is more of an implementation detail of launch_ros. It could also lead to misuse (e.g. I can imagine a user passing the node to their own executor).

I think it makes more sense for users to create their own nodes and manage their own contexts if they need them, and it doesn't involve too much overhead.

Maybe someone else can weigh in here.

Reply:
👍, keep the launch_ros context and node hidden, let the user create their own if they want to use ROS for something.

Reply:
I can see how this could lead to misuse, yeah. But on the other hand, we'd be disallowing any form of ROS context configuration, which was previously possible via [`ROSSpecificLaunchStartup`](https://github.com/ros2/launch_ros/blob/d10e4429563920fe101c4ce721f53eb3f1fe3b05/launch_ros/launch_ros/default_launch_description.py#L28). So I'd be personally inclined towards simply documenting this properly. 

One last thought. It may prove useful to allow users to create nodes and have launch servicing them and handling their lifetime.

Reply:
> One last thought. It may prove useful to allow users to create nodes and have launch servicing them and handling their lifetime.

We could provide that, but I think that's a different use case than this class. This class is useful (imo) for other actions to build on. If the we wanted to share something with the user and manage the lifetime for them, then we need a way to know when they are done with it and a way to give them access to the node and context we created. Though there is a lot of overlap, it seems like a different use case to me.

Reply:
Having the ability to configure the context seems compelling to me, but I wonder if there are other ways to allow that without exposing this to the user.

Reply:
Following @wjwwood rationale, if users won't be making use of this class for their own purposes then there's no need for the context to be configurable.